docs/docs/APIdocs/layers/optim/Adagrad.md

```diff
@@ -0,0 +1,60 @@
+## Adagrad ##
+
+**Scala:**
+```scala
+val adagrad = new Adagrad(var learningRate: Double = 1e-3,
+                          var learningRateDecay: Double = 0.0,
+                          var weightDecay: Double = 0.0)
+adagrad.optimize(feval, x, config, state)
```

Comment:
I think usually we only put a example of how to create an instance here. Only if there's some method that user really should know, we can also put it here. 

Is the `optimize` method user really needs to know? I thought in most uses a user just create an Adagrad instance and then set it into Optimzer as `optim_method`.  

I think we shoud remove this line.